src/backend/api/meals.js

```diff
@@ -2,14 +2,199 @@ const express = require("express");
 const router = express.Router();
 const knex = require("../database");
 
-router.get("/", async (request, response) => {
+
+router.get("/", async(request, response) => {
+
+    try {
+        // knex syntax for selecting things. Look up the documentation for knex for further info
+        const meals = await knex("meals");
+
+        if (request.query.hasOwnProperty('id')) {
+
+            const requestId = Number(request.query.id);
+            const requestedMeal = await knex('meals')
+                .select()
+                .where('id', '=', requestId);
+
+            console.log(requestedMeal);
+            console.log(typeof(requestedMeal));
+            if (Object.keys((requestedMeal)).length !== 0) {
+
+                response.json(requestedMeal);
+            } else {
+                response.status(404).json(`No meal matches the id: ${requestId}`);
+            }
+
+        } else if (Object.keys(request.query).length > 1 &&
+            request.query.hasOwnProperty('maxPrice', 'limit')) {
+
```

Comment:
`hasOwnProperty` takes one argument, so it will always ignore the "limit" here. 
That does not break the implementation, you are still establishing each separately and querying by it, which is very good 👌🏽  However, the way code lands in this condition is just by `maxPrice`, which might produce problems if `limit` is not in the query, as the existence of `limit` in the query is not checked here.